### PR TITLE
bump golangci-lint to v1.53.3

### PR DESCRIPTION
**Pull request by vrothberg**
Initially disable new linter (warnings).  Also fix some straight away as the warnings concerned me, especially the sql error checks.  Please refer to the individual commits for details.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -13,6 +13,17 @@ linters:
   enable-all: true
   disable:
     # All these break for one reason or another
+    - rowserrcheck # FIXME (urgent): sqlite error checking
+    # too many reports but requires attention
+    - depguard
+    - revive
```
[the remark this answers is not in the document]
Comment (vrothberg) [reply .golangci.yml:17]: Looks like you force me to spend more time than I wanted to on this linter thing ;-)

**Comment by vrothberg** [comment]
@Luap99 re-enabled and partially disabled :heavy_check_mark:

File: test/e2e/create_test.go [hunk 2]
```diff
@@ -315,7 +315,7 @@ var _ = Describe("Podman create", func() {
 	It("podman create --authfile with nonexistent authfile", func() {
 		session := podmanTest.Podman([]string{"create", "--authfile", "/tmp/nonexistent", "--name=foo", ALPINE})
 		session.WaitWithDefaultTimeout()
-		Expect(session).To(Not(Equal(0)))
+		Expect(session).ToNot(HaveValue(Equal(0)))
```
Comment (Luap99) [inline test/e2e/create_test.go:318]: This does not look right. 

I assume the test wants to check the exit code:
`Expect(session).To(Exit(125))`
The test is very broken for a long time: https://github.com/containers/podman/commit/d7c0f968ca60994306c8c76cd8e4e0a677fe9ada and thus failed to catch a regression,
compare https://github.com/containers/podman/pull/18413, for consistency this should error out as well.

File: test/e2e/play_kube_test.go [hunk 3]
```diff
@@ -3665,7 +3665,7 @@ o: {{ .Options.o }}`})
 		kube := podmanTest.Podman([]string{"play", "kube", kubeYaml})
 		kube.WaitWithDefaultTimeout()
 		if IsRemote() {
-			Expect(kube).Error()
+			Expect(kube).To(ExitWithError())
```
Comment (Luap99) [inline test/e2e/play_kube_test.go:3668]: can we check for the actual error code here not just any, https://github.com/containers/podman/issues/18188

**Review by Luap99 — COMMENTED** [review]
Thanks, two comments. And they really show how dangerous ginkgo test are because the test can be so easily written incorrectly and thus turn into a NOP.

File: test/e2e/create_test.go (hunk 2 above)
Comment (vrothberg) [reply test/e2e/create_test.go:318]: Not only the test is broken but also the code.  It should error but doesn't. I'll change the test but will also open an issue.

Comment (vrothberg) [reply test/e2e/create_test.go:318]: https://github.com/containers/podman/issues/18938

**Review by giuseppe — APPROVED** [review]
LGTM

File: test/e2e/play_kube_test.go [hunk 4]
```diff
@@ -3648,7 +3648,7 @@ o: {{ .Options.o }}`})
 		Expect(inspect.OutputToString()).To(ContainSubstring("o: " + volOpts))
 	})
 
-	It("podman play kube persistentVolumeClaim with source", func() {
+	FIt("podman play kube persistentVolumeClaim with source", func() {
```
Comment (Luap99) [inline test/e2e/play_kube_test.go:3651]: somebody is not using my new `FOCUS=...` Makefile option :(

File: test/e2e/create_test.go (hunk 2 above)
Comment (Luap99) [reply test/e2e/create_test.go]: yes that is what I meant with it failed to catch this regression.

File: test/e2e/play_kube_test.go (hunk 4 above)
Comment (vrothberg) [reply test/e2e/play_kube_test.go]: A good reminder to change my muscle memory and bash history

**Pull request merged**